### PR TITLE
blog: star history yearly pick-frontend 2022

### PR DESCRIPTION
**Pull request by milasuperstar**

### CODE REVIEW

File: public/blog/data.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
 [
+  {
+    "title": "Star History Yearly Pick | Top Front-end Projects 2022",
```
Comment (tianzhou) [inline public/blog/data.json:3]:
```suggestion
    "title": "Star History Yearly Pick 2022 | Front-end",
```

**Review by tianzhou — CHANGES_REQUESTED** [review]
https://star-history-frontend-prod-pr-271.onrender.com/blog/yearly-pick-frontend-2022

The preview does not render

File: public/blog/data.json [hunk 2]
```diff
@@ -1,4 +1,13 @@
 [
+  {
+    "title": "Star History Yearly Pick 2022 | Front-end",
+    "slug": "yearly-pick-frontend-2022",
```
Comment (boojack) [inline public/blog/data.json:4]:
```suggestion
    "slug": "star-history-yearly-pick-2022-frontend",
```

File: public/sitemap.xml [hunk 3]
```diff
@@ -9,6 +9,9 @@
 <url>
   <loc>https://star-history.com/blog</loc>
 </url>
+<url>
+  <loc>https://star-history.com/blog/yearly-pick-frontend-2022</loc>
```
Comment (boojack) [inline public/sitemap.xml:13]:
```suggestion
  <loc>https://star-history.com/blog/star-history-yearly-pick-2022-frontend</loc>
```

**Review by boojack — APPROVED** [review]
LGTM

**Review by adela-bytebase — APPROVED** [review]
LGTM

File: public/blog/star-history-yearly-pick-2022-frontend.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
+# Front-end projects of 2022
+
+As a proper goodbye to 2022, Star History looks at some of the most interesting open-source projects in the year that passed. Some projects came out of nowhere and made a big bang; some released new versions, while some renovated their websites.
+
+In today’s post, we are covering front-end projects.
+
+*Note: The following ranking is not in any particular order.
+
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-frontend.md:8]:
```suggestion
# Front-end projects of 2022

As a proper goodbye to 2022, Star History looks at some of the most interesting open-source projects in the year that passed. Some projects came out of nowhere and made a big bang; some released new versions, while some renovated their websites.

In today’s post, we are covering front-end projects.

```

Comment (tianzhou) [reply public/blog/star-history-yearly-pick-2022-frontend.md:8]: We are in alphabetical order

File: public/blog/star-history-yearly-pick-2022-frontend.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- simple, composable primitives without hidden rules and gotchas.
+- growing ecosystem.
+
+## Tailwind CSS
+
+https://github.com/tailwindlabs/tailwindcss
+
+![tailwind](/blog/assets/yearly-pick-frontend-2022/tailwind.webp)
+
+Tailwind CSS is a CSS framework toolset, and we use Tailwind for [bytebase.com](https://www.bytebase.com/). Whether it's colors or spacing, proportions, or fonts, Tailwind comes with defaults that are ready to use out of the box, but they can also be customized.
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-frontend.md:97]:
```suggestion
Tailwind CSS is a CSS framework toolset. Whether it's colors or spacing, proportions, or fonts, Tailwind comes with defaults that are ready to use out of the box, but they can also be customized.
```

**Review by tianzhou — APPROVED** [review]
LGTM

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**